### DISCUSSION TITLE
Modding extension, shaders, model viewer, QoL...

### DISCUSSION THREAD

**Discussion by snouz** [Albeoris/Memoria#653] [Releases]
![image](https://github.com/user-attachments/assets/a7caafae-f41a-433d-92db-9946cc8d951a)
[COMPLETE CHANGELOG HERE](https://github.com/Albeoris/Memoria/wiki/Changelog-v2024.08.05)

## Features
- 3 experimental **shaders for models** (Realistic, Toon, outlines) [#531](https://github.com/Albeoris/Memoria/pull/531) (by FakeEric)
- **Texture filtering** settings (allows square pixels like PSX)
- **Disc change** screens
- Qol: **Soft reset**, **minimap position**, visible inventory stock in synth shops
- Launcher changes:
  - Cheats: "manual trance", "10000 cards" and "lower TetraMaster randomness", simplified some options, check updates by default,
  - Adv. menu with model viewer shortcut

## Modding
- **Model Viewer overhaul**
- **All animated textures & 2D sfx are now moddable**
- Most textures (battle, npc, monsters, worldmap) are **moddable in folders** as png
- **Statuses moddable** (full overhaul)
- Any model can carry any model as weapon
- Export battle textures
- Many other possibilities (Trance models, SFX, custom textures, item/abilities, scan, scripts, white wind, Ncalc parameters, mix feature...)
- Mods: Moddb links supported, Memoria version check

## Bugfixes
- Settings: Force narrow screen if resolution too short, Steam overlay patcher removed if not windows, ini file more robust,
- Battle fixes: some attacks too slow, some double inputs, ViviAutoAttack, translations, AutoPotionOverhealLimit
- Script fixes: Instant quitting, Dagguerreo NPC, Desert palace widescreen spoil, turbo dialog
- Visual fixes:
  - 8 battle textures problems
  - 18+ bugged/missing animations, lights and shadows
  - 11 widescreen problems
- Thank you to @barkermn01 for signing this patcher

@Tirlititi @EricHu33 @DV666 @snouz @SamsamTS

<hr /><em>This discussion was created from the release <a href='https://github.com/Albeoris/Memoria/releases/tag/v2024.08.05'>Modding extension, shaders, model viewer, QoL...</a>.</em>

**Comment by ElElixir** [comment]
Wow, these new characters models look pretty good! Good work!

**Reply by barkermn01** [reply]
They are not new model's the difference is there is now a lighting engine and using shaders something that was not possible on the PS1 so was not added to the SE Unity port, @EricHu33 has added support for Shaders with this.

On a Side note i do hope with this system in place that people will now work on a mod with new models in that will really show off the shader support.

**Reply by ElElixir** [reply]
Ok, didnt know that.
Battle models looks like they are new,  so it is a pretry good work.
Im enjoying it a lot 
See ya

**Comment by WarpedEdge** [comment]
Waiting for DV's tutorial on character adds, then I may expand on it in
depth for "Characters existing that have battle models" & "Characters
existing but do not have battle models" and adding weapons to each, or
adding animations for battle or overworld characters. My hope is to make
every NPC/Enemy normal in size (unless i can shrink bigger enemies) to be
playable in functionality (skills and stuff is separate). If I can make a
few characters like this, I am hoping I can either automate the creation of
new characters if they all follow the same system. THEN I hope someone
makes it possible to inject NEW models into the game so people can have
custom characters and whatnot be playable.

**Comment by NyxFF** [comment]
Can I ask if these features are included in the 9.0 version which is available on the official site or is it the changelog for a future patch? The Disc Change screen for example (thank you so much, I've been asking for it since 2016 !) isn't available in 9.0 as far as I know.

Thanks;)

**Reply by SamsamTS** [reply]
By version 9.0 I'm assuming you are talking about Moguri. Moguri contains 0 features has it is a texture mod and only contains (beautifully upscaled) textures.

All the features come from Memoria, if you installed this version then you get all those features, including disc change screens.

**Reply by NyxFF** [reply]
Ah I see, the option was hidden in the .ini file, that's why I couldn't figure out if the feature existed. Do you also know what TileSize is and whether I should have it on 32 or 64? It's set to 32 by default although it says Moguri Mod is 64.
Lastly, (since you're here^^) should I use bilinear or trilinear for all the SmoothTexture options? 

Thanks

**Reply by SamsamTS** [reply]
Yeah the disc screen option will be exposed in the new launcher coming next update (soon™).
I recommend leaving TileSize to 32. Moguri will automatically force it to 64 when the mod is enabled.

I personally don't notice any discernible change between Bilinear and Trilinear

**Comment by charlie-in-spain** [comment]
Hi! 

I've tried to edit Mini status in "StatusData.csv" file so that it has the same colours as Berserk status, in order to try and see how this works, but this won't apply in battles.  
I've set ColorKind to "0" as you can see, and also "ColorBase":

![statusdata](https://github.com/user-attachments/assets/9fb01165-5be6-4fd5-bb7c-36aa793db1e5)

Do I need to do something else?

Thanks in advance!

**Reply by Tirlititi** [reply]
Memoria CSV files are better modified with text editors like Notepad or Notepad++, because spreadsheet softwares tend to change their formatting. I copy/paste a message I just wrote to someone who had a similar problem:

The thing is, CSV is a very old and very flexible file format. The CSV formatting we use has several "valid but unusual" specifications:
1) It uses semi-colons as field separators, instead of commas. This is supported by many CSV reader at least when importing the files (but sometimes semi-colons are replaced by commas when overwriting the file).
2) It uses comments for lines starting with the symbol ``#``. Again, this is supported in reading by many softwares that read CSV: if they see lines starting with that symbol, they skip it. But then, Memoria uses some of these commented lines to give actual informations about the version of the CSV we use (the ``#! Include...`` lines). So it is important that these comments are kept, and Excel-like tools tend to just discard them.

So if you find options that address 1) and 2), then surely you can use Excel. Something like "Use semi-colons as field separators" and "Keep comment lines". I'm not sure they exist though

**Reply by charlie-in-spain** [reply]
Hi Tirlititi, 

I edited "StatusData.csv" with Notepad and it worked! So thanks a lot for your solution. 

Regards!
